### PR TITLE
feat(css-parser): support selectors

### PR DESCRIPTION
**Pull request by yeonjuan**
## Summary
part of #984

This pr includes the implementation of parsing some CSS selectors.
It adds new CSS AST nodes, parsing, and formatting.

- Selectors
  -  CSSClassSelector : `.class`
  -  CSSIdSelector: `#id`
  - CSSTypeSelector: `type`
  - CSSPseudoClassSelector: `:class`
  - CSSPseudoElementSelector: `::element`
 - Combinators
   - descendant: ` `
   - child: `>`
   - nextSibiling: `+`
   - subsequentSibiling: `~`

## Test Plan

test case

### CODE REVIEW

File: internal/css-parser/parser/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			patterns.push(combinator);
+		}
+
+		if (selector) {
+			patterns.push(selector);
+		}
+
+		if (!selector && !combinator) {
+			parser.unexpectedDiagnostic({
+				description: descriptions.CSS_PARSER.UNEXPECTED_TOKEN,
```
Comment (ematipico) [inline internal/css-parser/parser/selectors.ts:166]: Do you think it's possible to have a better error message here? Here we know what could be the issue, so emitting a more describing message would help to understand the cause of the problem.

File: internal/css-parser/test-fixtures/selectors/input.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+div, div + #id, div ~ #id, div > #id {
```
Comment (ematipico) [inline internal/css-parser/test-fixtures/selectors/input.css:1]: We should also add different cases, bad ones included. I try to suggest something:

```css
// this should be a parsing error
>, { 

}
```
```css
// this should work
span ~ + {} 
``` 
```css
+ {}
```

File: internal/formatter/test-fixtures/css/selectors/input.test.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+div + #id {
+}
+
+div ~ #id {
+}
+
+div > #id {
+}
+
+div + #id, div ~ #id, div > #id {
```
Comment (ematipico) [inline internal/formatter/test-fixtures/css/selectors/input.test.md:91]: Now that we are starting to have something, I think we should start making some decisions about the CSS formatting.

Personally, I like having something like:


```css
div + #id, 
div ~ #id, 
div > #id {

}
```

This is my personal opinion. We can have a chat in some issue. What do you think?

File: internal/formatter/test-fixtures/css/selectors/input.test.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
```
Comment (ematipico) [inline internal/formatter/test-fixtures/css/selectors/input.test.md:50]: We should add a text case with long strings and see how the formatter behaves when it needs to go to a new line.

**Review by ematipico — COMMENTED** [review]
Overall it looks good do me. I would add some bad case inside the parser in order to try to catch possible error and make them more descriptive. I would also try to add some edge case around the formatter

File: internal/formatter/test-fixtures/css/selectors/input.test.md (hunk 3 above)
Comment (yeonjuan) [reply internal/formatter/test-fixtures/css/selectors/input.test.md]: @ematipico cool, make sense! it's actually what prettier does.

File: internal/css-parser/test-fixtures/selectors/input.css (hunk 2 above)
Comment (yeonjuan) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: @ematipico Thanks :) I added a parsing error case also.

> // this should work
> span ~ + {} 
> + {}

As far as I know, these are invalid syntax right?

Comment (ematipico) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: Yeah my bad. The ones that I mentioned are invalid syntax.

These should be valid syntax

```css
.field + * {}
.field ~ * {}
.field * {}
```

Comment (yeonjuan) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: @ematipico Yes Thanks for the reply. Currently, This PR didn't contain parsing selector (`*`) implementation. I'll add that soon. :)

Comment (antonk52) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: It would be nice to add a test for `.long\.class\.name`, apparently class names can contain dots which should be escaped using `\` in css ie `.long\.class\.name {}`. Even though it's quite an edge case I think it would be nice to test against it.

Comment (ematipico) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: @yeonjuan  yes let's add the universal selector with a different PR. It makes sense! (sorry I thought it was already supported, my bad).

Comment (yeonjuan) [reply internal/css-parser/test-fixtures/selectors/input.css:1]: @antonk52 
> It would be nice to add a test for .long\.class\.name, apparently class names can contain dots which should be escaped using \ in css ie .long\.class\.name {}. Even though it's quite an edge case I think it would be nice to test against it.

Sure. It makes sense. 👍  Thanks

**Comment by yeonjuan** [comment]
@ematipico Hi Thanks for the reviewsI add descriptive error messages but not sure it's enough. Please give me feedback when you have time :)

**Review by ematipico — APPROVED** [review]
That's great! Let me know when you rebase the PR so we can merge it

**Comment by yeonjuan** [comment]
@ematipico Thanks for the review! There was a minor bug in #1287 about the comment position and I added to fix the issue also - [code](https://github.com/rome/tools/pull/1281/commits/adb4b605f147fc2ffabf7dcb086424e8203b95a6)
I think It can be merged now Thanks :)

**Comment by ematipico** [comment]
Cool! I noticed we still have some things to manage.

A style like this throws an error:

```css
.style {
	font-family: "Fira Code", monospace;
}
```

We don't parse the `,` after `"Fira Code"`.

**Comment by yeonjuan** [comment]
@ematipico Yes you are right 👍  Could I work on that?

**Comment by ematipico** [comment]
Sure! Thank you!!
